crates/api_models/src/admin.rs

```diff
@@ -163,33 +163,101 @@ pub struct MerchantConnectorId {
     pub merchant_id: String,
     pub merchant_connector_id: i32,
 }
-//Merchant Connector Account CRUD
-#[derive(Debug, Clone, Serialize, Deserialize)]
+
+/// Create a new Payment Connector for the merchant account. The connector could be a payment processor / facilitator / acquirer or specialized services like Fraud / Accounting etc."
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
 #[serde(deny_unknown_fields)]
 pub struct PaymentConnectorCreate {
+    /// Type of the Connector for the financial use case. Could range from Payments to Accounting to Banking.
+    #[schema(example = "payment_processor,payment_vas")]
     pub connector_type: api_enums::ConnectorType,
+    /// Name of the Connector
+    #[schema(value_type = String,example = "stripe,adyen")]
     pub connector_name: String,
+    /// Unique ID of the connector
+    #[schema(value_type = Option<i32>,example = 42)]
     pub merchant_connector_id: Option<i32>,
+    /// Account details of the Connector. You can specify up to 50 keys, with key names up to 40 characters long and values up to 500 characters long. Useful for storing additional, structured information on an object.
+    #[schema(value_type = Option<Object>,example = json!({ "auth_type": "HeaderKey","api_key": "Basic MyVerySecretApiKey" }))]
     pub connector_account_details: Option<Secret<serde_json::Value>>,
+    /// A boolean value to indicate if the connector is in Test mode. By default, its value is false.
+    #[schema(value_type = Option<bool>,default = false, example = false)]
     pub test_mode: Option<bool>,
+    /// A boolean value to indicate if the connector is disabled. By default, its value is false.
+    #[schema(value_type = Option<bool>,default = false, example = false)]
     pub disabled: Option<bool>,
+    /// Refers to the Parent Merchant ID if the merchant being created is a sub-merchant
+    #[schema(value_type = Option<Vec<PaymentMethods>>,example = json!([
```

Comment:
Explicitly specifying `value_type` for `Option<bool>` and `Option<Vec<PaymentMethods>>` is not required.